deep-security-agent/recipes/default.rb

```diff
@@ -10,6 +10,14 @@
 # * Install the Deep Security Agent
 # *********************************************************************
 
+if node['ec2'] && node['ec2']['instance_type'] != 't2.small' && node['deep_security_agent']['action'] == 'install'
```

Comment:
Since we're overriding some requested behavior here (action is set to install, but won't do if instance type meets certain criteria) I think it would be prudent to print out some sort of message that we're overriding the user's choice. Then folks aren't confused later when the agent doesn't get installed even though they have said they ought to.